dubbo-admin-backend/src/test/java/org/apache/dubbo/admin/controller/LoadBalanceControllerTest.java

```diff
@@ -0,0 +1,147 @@
+        assertEquals(HttpStatus.CREATED, response.getStatusCode());
+        assertTrue(Boolean.valueOf(response.getBody()));
+    }
+
+    @Test
+    public void updateLoadbalance() throws IOException {
+        String id = "1";
+        BalancingDTO balancingDTO = new BalancingDTO();
+        URI uri;
+        ResponseEntity<String> response;
+        // unknown id
+        uri = restTemplate.getUriTemplateHandler().expand(url("/api/{env}/rules/balancing/{id}"), env, id);
+        response = restTemplate.exchange(uri, HttpMethod.PUT, new HttpEntity<>(balancingDTO), String.class);
+        assertFalse("should return a fail response, when id is null", (Boolean) objectMapper.readValue(response.getBody(), Map.class).get("success"));
+        // valid id
+        BalancingDTO balancing = mock(BalancingDTO.class);
+        when(overrideService.findBalance(id)).thenReturn(balancing);
+        uri = restTemplate.getUriTemplateHandler().expand(url("/api/{env}/rules/balancing/{id}"), env, id);
+        response = restTemplate.exchange(uri, HttpMethod.PUT, new HttpEntity<>(balancingDTO), String.class);
+        assertTrue(Boolean.valueOf(response.getBody()));
+        verify(overrideService).saveBalance(any(BalancingDTO.class));
+    }
+
+    @Test
+    public void searchLoadbalances() throws IOException {
+        String service = "test service", application = "test application";
+        ResponseEntity<String> response;
+        // service and application are all blank
+        response = restTemplate.getForEntity(url("/api/{env}/rules/balancing"), String.class, env);
+        assertFalse("should return a fail response, when service and application are all blank", (Boolean) objectMapper.readValue(response.getBody(), Map.class).get("success"));
+        // service is valid
+        response = restTemplate.getForEntity(url("/api/{env}/rules/balancing?service={service}&application={application}"), String.class, env, service, null);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        verify(overrideService).findBalance(service);
+        // application is valid
+        response = restTemplate.getForEntity(url("/api/{env}/rules/balancing?service={service}&application={application}"), String.class, env, null, application);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        verify(overrideService).findBalance(application);
+        // findBalance return a notnull
+        BalancingDTO balancingDTO = new BalancingDTO();
+        when(overrideService.findBalance(anyString())).thenReturn(balancingDTO);
+        response = restTemplate.getForEntity(url("/api/{env}/rules/balancing?service={service}&application={application}"), String.class, env, null, application);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertEquals(1, objectMapper.readValue(response.getBody(), List.class).size());
```

Comment:
Why not specify the return type as List.class? Then there is no need to use ObjectMapper, you can use responseEntity.getBody()